### ISSUE TITLE
Unable to run BmiClientSingularity from inside Jupyter notebook

### ISSUE THREAD

**Issue by sverhoeven** [eWaterCycle/grpc4bmi#56]
Running a BmiClientSingularity inside a notebook causes a exception about the stdout not having a fileno() method.

The BmiClientSingularity ties the stdout/stderr of the Singularity container to the Python client process.
Jupyter notebook replaces it's stdout/stderr with a zeromq pipe which can not be tied to a subprocess.

**Comment by sverhoeven** [comment]
Reproduced by taking
https://github.com/eWaterCycle/notebooks/blob/master/bmi/PCR-GLOBWB.ipynb notebook and replacing BmiClientDocker construction with
```python
# Startup model
from grpc4bmi.bmi_client_singularity import BmiClientSingularity
model = BmiClientSingularity(image='/mnt/apps/singularity-images/ewatercycle-pcrg-grpc4bmi.sif', input_dir="./input", output_dir="./output")
```

Will raise the following exception:
```python
---------------------------------------------------------------------------
UnsupportedOperation                      Traceback (most recent call last)
<ipython-input-11-15edc2213eac> in <module>
      2 model = BmiClientSingularity(image='/mnt/apps/singularity-images/ewatercycle-pcrg-grpc4bmi.sif',
      3                         input_dir="./input",
----> 4                         output_dir="./output")

/usr/local/lib/python3.6/dist-packages/grpc4bmi/bmi_client_singularity.py in __init__(self, image, input_dir, output_dir)
     69         env['BMI_PORT'] = str(port)
     70         logging.info(f'Running {image} singularity container on port {port}')
---> 71         self.container = subprocess.Popen(args, stderr=sys.stderr, stdout=sys.stdout, env=env, preexec_fn=os.setsid)
     72         super(BmiClientSingularity, self).__init__(BmiClient.create_grpc_channel(port=port, host=host))
     73 

/usr/lib/python3.6/subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, encoding, errors)
    685         (p2cread, p2cwrite,
    686          c2pread, c2pwrite,
--> 687          errread, errwrite) = self._get_handles(stdin, stdout, stderr)
    688 
    689         # We wrap OS handles *before* launching the child, otherwise a

/usr/lib/python3.6/subprocess.py in _get_handles(self, stdin, stdout, stderr)
   1202             else:
   1203                 # Assuming file-like object
-> 1204                 c2pwrite = stdout.fileno()
   1205 
   1206             if stderr is None:

UnsupportedOperation: fileno
```

**Comment by sverhoeven** [comment]
Solved by #58 PR
